cinema/serializers.py

```diff
@@ -68,13 +79,49 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available",
         )
 
 
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = SerializerMethodField()
 
     class Meta:
         model = MovieSession
-        fields = ("id", "show_time", "movie", "cinema_hall")
+        fields = ("id", "show_time", "movie", "cinema_hall", "taken_places")
+
+    def get_taken_places(self, movie_session):
+        return [{"row": ticket.row, "seat": ticket.seat}
+                for ticket in movie_session.tickets.all()]
+
+
+class TicketSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = ("id", "row", "seat", "movie_session")
+
+
+class TicketRetrieveSerializer(TicketSerializer):
+    movie_session = MovieSessionListSerializer(many=False, read_only=True)
```

Comment:
False is a default value, thus you can remove it

Reply:
Good markm change this please